prog.go

```diff
@@ -413,6 +413,16 @@ func (p *Program) MarshalBinary() ([]byte, error) {
 	return buf, nil
 }
 
+// Attach a Program to a container object fd
+func (p *Program) Attach(fd uint32, bfpProgType ProgramType, flags uint32) error {
```

Comment:
`fd` should be `int`, and then we should check for `fd < 0`, to be consistent with other fds in the API. I think `flags` could be `type AttachFlags uint32`. We don't need to define all the flags now, but we leave this option open to us. User code can do `foo = ebpf.AttachFlags(0xbar)` in the meantime.

I'm a bit fuzzy what `bpfProgType` is for. Shouldn't it be `AttachType`?

Reply:
You are correct, it should be `AttachType` - sorry for the mistake!